packages/application-config/src/load-deprecated-config.ts

```diff
@@ -0,0 +1,67 @@
+import type { ApplicationConfig } from './types';
+
+import fs from 'fs';
+import substituteEnvVariablePlaceholders from './substitute-env-variable-placeholders';
+
+export type DeprecatedOptions = {
+  envPath: string;
+  headersPath: string;
+  cspPath?: string;
+};
+
+// List the required fields of `env.json`
+const requiredEnvJsonFields = [
+  'applicationName',
+  'frontendHost',
+  'mcApiUrl',
+  'location',
+  'env',
+  'cdnUrl',
+];
+
+const parseJsonFile = (filePath: string) => {
+  let rawData;
+  try {
+    rawData = fs.readFileSync(filePath, {
+      encoding: 'utf8',
+    });
+  } catch (error) {
+    // Ignore
+  }
+  return rawData ? JSON.parse(rawData) : {};
+};
+
+// For backwards compatibility, we still support the `env.json` and `headers.json` files.
+// TODO: remove in `v17`.
+const loadDeprecatedConfig = (
```

Comment:
This is the part where we load the `env.json` and `headers.json` for backwards compatibility